### PR TITLE
geo/wkt: add support for parsing polygons with Z and M dimensions

### PR DESCRIPTION
**Pull request by andyyang890**
This patch extends the capabilities of the WKT parser to include
parsing of polygons with Z and M dimensions.

Refs: #53091 

Release note: None

### CODE REVIEW

File: pkg/geo/wkt/wkt.y [hunk 1]
```diff
@@ -14,25 +14,57 @@ package wkt
 
 import "github.com/twpayne/go-geom"
 
+func isValidLinestring(wktlex wktLexer, flatCoords []float64, stride int) bool {
```
Comment (otan) [inline pkg/geo/wkt/wkt.y:17]: nit: `LineString`

File: pkg/geo/wkt/wkt_test.go [hunk 2]
```diff
@@ -169,6 +236,34 @@ POINT(0, 0)
...
+			desc:  "2D polygon with ring that isn't closed",
+			input: "POLYGON((0 0, 1 1, 2 0, 1 -1))",
+			expectedErrStr: `syntax error: polygon ring not closed at pos 28
+POLYGON((0 0, 1 1, 2 0, 1 -1))
+                            ^`,
+		},
+		{
+			desc:  "2D polygon with empty ring",
+			input: "POLYGON((0 0, 1 -1, 2 0, 0 0), ())",
+			expectedErrStr: `syntax error: unexpected ')', expecting NUM at pos 32
```
Comment (otan) [inline pkg/geo/wkt/wkt_test.go:256]: can you add a test for an invalid ring?

**Review by otan — APPROVED** [review]
very nice

File: pkg/geo/wkt/wkt_test.go (hunk 2 above)
Comment (andyyang890) [reply pkg/geo/wkt/wkt_test.go:256]: Added

**Comment by andyyang890** [comment]
bors r=otan

**Pull request merged**